gensec.py

```diff
@@ -55,19 +62,29 @@
     if not os.path.exists(parameters["calculator"]["optimize"]):
         os.mkdir(parameters["calculator"]["optimize"])
     os.chdir(parameters["calculator"]["optimize"])
-    output = Output(os.path.join(os.getcwd(), "report_{}.out".format(parameters["calculator"]["optimize"])))
+    output = Output(
+        os.path.join(
+            os.getcwd(), "report_{}.out".format(parameters["calculator"]["optimize"])
+        )
+    )
     dirs.find_last_dir(parameters)
     known.analyze_calculated(structure, fixed_frame, parameters)
     # output.write_parameters(parameters, structure, known, dirs)
     # for f in glob.glob("/tmp/ipi_*"):
     #     if os.path.exists(os.path.join("/tmp/", f)):
     #         os.remove(os.path.join("/tmp/", f))
-    
+
     structure.mu = np.abs(calculator.estimate_mu(structure, fixed_frame, parameters))
     # Finish unfinished calculation
-    calculator.finish_relaxation(structure, fixed_frame, parameters, dirs.current_dir(parameters))
+    calculator.finish_relaxation(
+        structure, fixed_frame, parameters, dirs.current_dir(parameters)
+    )
     workflow.success = dirs.dir_num
-    generated_dirs = [z for z in os.listdir(dirs.generate_folder) if os.path.isdir(os.path.join(dirs.generate_folder, z))]
+    generated_dirs = [
+        z
+        for z in os.listdir(dirs.generate_folder)
+        if os.path.isdir(os.path.join(dirs.generate_folder, z))
+    ]
 
     # while len(generated_dirs)>0:
```

Comment:
Large portions of commented code should be cleaned out. if one really wants to save it for future purposes, do it in a branch.

Reply:
All the routines for generation and optimization rewritten and moved to the protocols.py
The main gensec.py will be much shorter.